syntax = "proto3";

package proto;

service SubscribeService {
  rpc Subscribe(SubscribeRequest) returns (stream SubscribeResponse) {}
  rpc Publish(PublishRequest) returns (PublishResponse) {}
}

message SubscribeRequest {
  string topic = 1;
  string id = 2;
}

message SubscribeResponse { string message = 1; }

message PublishRequest {
  string topic = 1;
  string message = 2;
}

message PublishResponse { string message = 1; }
